### PR TITLE
UI revamp

### PR DESCRIPTION
**Pull request by v0idzz**
This adds docking system to the app and improves the layout and overall look of the UI. With these changes, the UI follows host OS theme.

https://github.com/user-attachments/assets/f3de511f-caba-42bc-952e-b7b82df7f85c

### CODE REVIEW

File: Reemit.Gui/Common/IconThemingUtility.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        var haloLuminosity = Color.FromArgb(255, 246, 246, 246).ToHsl().L;
+
+        if (backgroundLuminosity < 0.5)
+        {
+            haloLuminosity = 1.0 - haloLuminosity;
+            luminosity = 1.0 - luminosity;
+        }
+
+        if (luminosity < haloLuminosity)
```
Comment (John-Leitch) [inline Reemit.Gui/Common/IconThemingUtility.cs]: Perhaps a little contentious, but we could fold this into one `return` e.g.

```
return luminosity < haloLuminosity ?
    backgroundLuminosity * luminosity / haloLuminosity :
    (1.0 - backgroundLuminosity) * (luminosity - 1.0) / (1.0 - haloLuminosity) + 1.0;
```

File: Reemit.Gui/Views/Controls/Icons/Icon.axaml.cs [hunk 2]
```diff
@@ -49,8 +78,22 @@ private void UpdateIcon()
...
+        AdjustIconLuminosity(icon);
+
         Children.Clear();
         Children.Add(icon);
     }
+
+    private void AdjustIconLuminosity(Control icon)
+    {
+        // This looks hack-ish, but works for all the VS icons we currently have.
+        if (icon.GetLogicalChildren().FirstOrDefault()?.GetLogicalChildren().FirstOrDefault() is Rectangle iconRectangle)
```
Comment (John-Leitch) [inline Reemit.Gui/Views/Controls/Icons/Icon.axaml.cs]: Somewhat of a nitpick, but should this be rewritten to be more resilient against changes in the XAML? Preferrably, an approach that doesn't expect children at specific offsets. Not sure what the right level of genericism is here, but perhaps we could search for every `SolidColorBrush` in the control and adjust luminosity. I suppose it depends on whether we expect to ever have a control that contains a `SolidColorBrush` we don't want to transform.

Also, a bit more out there and quite out of scope for this PR, but: I wonder if we could find a way to move some of this to the XAML. Maybe a markup extension or something. However, probably not worth doing if we don't expect to transform luminosity outside of this.

**Review by John-Leitch — APPROVED** [review]
Looks and works great. Awesome to see it coming together, the docking functionality is slick. Only a couple nitpicky comments.

File: Reemit.Gui/Common/IconThemingUtility.cs (hunk 1 above)
Comment (v0idzz) [reply Reemit.Gui/Common/IconThemingUtility.cs]: Done.

File: Reemit.Gui/Views/Controls/Icons/Icon.axaml.cs (hunk 2 above)
Comment (v0idzz) [reply Reemit.Gui/Views/Controls/Icons/Icon.axaml.cs]: Good idea. Rewrote this to work regardless of the layout of the control tree.

**Pull request merged**